frontend/public/components/cloud-services/subscription.tsx

```diff
@@ -20,98 +31,189 @@ const subscriptionState = (state: SubscriptionState) => {
+export const SubscriptionsPage: React.SFC<SubscriptionsPageProps> = (props) => <ListPage
   {...props}
+  kind={referenceForModel(SubscriptionModel)}
   title="Subscriptions"
   showTitle={true}
   canCreate={true}
-  createProps={{to: props.match.params.ns ? `/k8s/ns/${props.match.params.ns}/${CatalogSourceModel.plural}` : `/k8s/all-namespaces/${CatalogSourceModel.plural}`}}
+  createProps={{to: props.namespace ? `/k8s/ns/${props.namespace}/${CatalogSourceModel.plural}` : `/k8s/all-namespaces/${CatalogSourceModel.plural}`}}
   createButtonText="New Subscription"
   ListComponent={SubscriptionsList}
-  filterLabel="Subscriptions by package"
-  flatten={resources => _.flatMap(_.filter(resources, (v, k: string) => k === referenceForModel(SubscriptionModel)), (resource: any) => resource.data)}
-  resources={[
-    {kind: referenceForModel(SubscriptionModel), isList: true, namespaced: true},
-    {kind: referenceForModel(ClusterServiceVersionModel), isList: true, namespaced: true},
-  ]} />;
+  filterLabel="Subscriptions by package" />;
 
-export type SubscriptionDetailsProps = {
-  obj: SubscriptionKind;
-  installedCSV: ClusterServiceVersionKind;
+export const SubscriptionDetails: React.SFC<SubscriptionDetailsProps> = (props) => {
+  const {obj, pkg, installedCSV} = props;
+
+  return <div className="co-m-pane__body">
+    <div className="co-m-pane__body-group">
+      <SubscriptionUpdates pkg={pkg} obj={obj} installedCSV={installedCSV} />
+    </div>
+    <div className="co-m-pane__body-group">
+      <div className="row">
+        <div className="col-sm-6">
+          <ResourceSummary resource={obj} showNodeSelector={false} showPodSelector={false} showAnnotations={false} />
+        </div>
+        <div className="col-sm-6">
+          {/* TODO(alecmerdler): Show if InstallPlan needs approval */}
+          <dl className="co-m-pane__details">
+            <dt>Installed Version</dt>
+            <dd>
+              { _.get(obj.status, 'installedCSV') && !_.isEmpty(installedCSV)
+                ? <ResourceLink kind={referenceForModel(ClusterServiceVersionModel)} name={_.get(obj.status, 'installedCSV')} namespace={obj.metadata.namespace} title={_.get(obj.status, 'installedCSV')} />
+                : 'None' }
+            </dd>
+            <dt>Starting Version</dt>
+            <dd>{obj.spec.startingCSV || 'None'}</dd>
+            <dt>Catalog</dt>
+            <dd>
+              <ResourceLink kind={referenceForModel(CatalogSourceModel)} name={obj.spec.source} namespace={obj.metadata.namespace} title={obj.spec.source} />
+            </dd>
+          </dl>
+        </div>
+      </div>
+    </div>
+  </div>;
 };
 
+export class SubscriptionUpdates extends React.Component<SubscriptionUpdatesProps, SubscriptionUpdatesState> {
+  constructor(props) {
+    super(props);
+    this.state = {waitingForUpdate: false, installPlanApproval: props.obj.spec.installPlanApproval, channel: props.obj.spec.channel};
```

Comment:
What do you think about being more careful here with the lookups since this class is exported?